backend/src/app.js

```diff
@@ -2,23 +2,43 @@ const express = require('express')
 const mongoose = require('mongoose')
 const dotenv = require('dotenv')
 
+var cors = require('cors')
+
+const studentsRouter = require('./routes/Students')
+const authRouter = require('./routes/Auth')
+
 const app = express()
 dotenv.config()
 
-mongoose.connect(process.env.MONGO_URI, {
-    // connect to mongoDB
-    useNewUrlParser: true,
-    useUnifiedTopology: true,
-})
+if (process.env.NODE_ENV === 'test') {
```

Comment:
can we change 'test' into 'stag' for staging (the other option which we can also explicitly define in the else statement is 'prod' for production)

Reply:
we should also change the database name in MongoDB appropriately

Reply:
Okay, so I will make the appropriate changes to API endpoints so that only the request with the required API key in the header can make the request. That will be stored privately in the app, I guess ?. 

Regarding line 13 in src/app.js -> I can't change the test option to stag because JEST automatically sets NODE_ENV='test' whenever tests are executed. I need that to test API endpoints on the test database rather than the actual database. If there is a workaround, I will look into it and make the changes if possible.

Thank you for reviewing the work and replying so quickly. 👍🏽

Reply:
> we should also change the database name in MongoDB appropriately

Yes, we can change the database name in MongoDB. Just let me know what should I name them or naming scheme should I follow

Reply:
> Okay, so I will make the appropriate changes to API endpoints so that only the request with the required API key in the header can make the request. That will be stored privately in the app, I guess ?.

Yes, it will be used from the environment variable

Reply:
> Regarding line 13 in src/app.js -> I can't change the test option to stag because JEST automatically sets NODE_ENV='test' whenever tests are executed. I need that to test API endpoints on the test database rather than the actual database. If there is a workaround, I will look into it and make the changes if possible.

I see, that's a misunderstanding on my side; I didn't think "test" is for automated tests. When we deploy the backend, we  want to have 2 version: staging and production. The staging one will be used for our development, while the production one for actual use by users. We can set different MongoDB URI similar to how we differentiate for testing

Reply:
staging will be connected to our develop branch while production will be connected to our main branch

Reply:
Sure we can do that.

Reply:
I have added the code block that will connect the backend to different databases based on the environment variable version provided to it.